wagtail/admin/views/generic/history.py

```diff
@@ -24,10 +24,16 @@
 )
 
 
+def get_actions_for_filter():
+    # Only return those actions used by model log entries.
+    actions = set(ModelLogEntry.objects.all().get_actions())
```

Comment:
Models won't necessarily log to ModelLogEntry, as per https://docs.wagtail.org/en/stable/extending/audit_log.html#log-models. I think this should be calling `log_registry.get_log_model_for_model` for the relevant model.

Reply:
Digging into this, it seems that passing model here is a bit fiddly, and the scenarios in which ModelLogEntry doesn't give the right results are pretty obscure, so I'm happy to let this slide to get it into 5.2.

(For the future, I think the smarter solution would actually be to limit the options to the ones actually seen in the current queryset, which sidesteps this issue...)

Reply:
Thanks @gasman!

Good point (future note), that could also be used to filter the visible users in the user-filter too.